src/domain_decomposition.hpp

```diff
@@ -0,0 +1,116 @@
+#pragma once
+
+#include <vector>
+
+#include <backends.hpp>
+#include <common_types.hpp>
+#include <communication/global_policy.hpp>
+#include <recipe.hpp>
+#include <util/optional.hpp>
+#include <util/partition.hpp>
+#include <util/transform.hpp>
+
+namespace nest {
+namespace mc {
+
+// Meta data used to guide the domain_decomposition in distributing
+// and grouping cells.
+struct group_rules {
```

Comment:
This contains information required to perform the domain decomposition. It doesn't require cell type information: such information is needed in the output of the domain decompositon, where it is used to build the concrete cell groups.